### PR TITLE
bpo-40417: Fix deprecation warning in PyImport_ReloadModule

### PR DESCRIPTION
**Pull request by Robmaister**
I can add another commit with the new test case I wrote to verify that the warning was being printed before my change, stopped printing after my change, and that the function does not return null after my change.

https://bugs.python.org/issue40417

https://bugs.python.org/issue40417

Automerge-Triggered-By: @brettcannon

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@Robmaister

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by Robmaister** [comment]
I emailed help@python.org about this issue, but I am unable to add my Github username to my BPO account due to what I can only assume is a bug.

I received the same error message when attempting to verify my email address when I first tried to make an account directly on BPO.

image

I got past that error by signing in with Google. Attempted to add my Github account on both Chrome 81.0.4044.122 and Internet Explorer 11.778.18362.0. Not sure what else I can try.

**Comment by Robmaister** [comment]
After a little bit of confusion, I was able to figure out what was going on with an error message I got when trying to set some of my info in BPO.

Github account linked, CLA was signed about an hour ago so everything should be good in about a day 😄

File: Python/import.c [hunk 1]
```diff
@@ -1977,16 +1977,16 @@ PyImport_ImportModuleLevel(const char *name, PyObject *globals, PyObject *locals
 PyObject *
 PyImport_ReloadModule(PyObject *m)
 {
-    _Py_IDENTIFIER(imp);
+    _Py_IDENTIFIER(importlib);
     _Py_IDENTIFIER(reload);
     PyObject *reloaded_module = NULL;
-    PyObject *imp = _PyImport_GetModuleId(&PyId_imp);
+    PyObject *imp = _PyImport_GetModuleId(&PyId_importlib);
```
Comment (ericsnowcurrently) [inline Python/import.c]:
```suggestion
    PyObject *importlib = _PyImport_GetModuleId(&PyId_importlib);
```

**Review by ericsnowcurrently — APPROVED** [review]
Mostly LGTM.  Renaming the variable to "importlib" would be better for readers.  I'll approve the PR but please consider making that change anyway. :smile:

I wouldn't worry about adding a test.  We don't really have a pattern (of which I'm aware at least) of verifying that the C-API does not emit deprecation warnings.

File: Python/import.c (hunk 1 above)
Comment (Robmaister) [reply Python/import.c]: Done!

**Comment by brettcannon** [comment]
@Robmaister mind adding a news entry? If you want to use a web form you can follow the "Details" link for the failing bedevere/news status check.

File: Misc/NEWS.d/next/Core and Builtins/2020-05-01-19-04-52.bpo-40417.Sti2lJ.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix "imp" module deprecation warning when PyImport_ReloadModule is called
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-01-19-04-52.bpo-40417.Sti2lJ.rst]:
```suggestion
Fix imp module deprecation warning when PyImport_ReloadModule is called. Patch by Robert Rouhani.
```

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
@Robmaister I made sure to thank you in the news entry. Once CI passes this should get auto-merged.

**Pull request merged**